### ISSUE TITLE
/usr/bin/env: 'python': No such file or directory (Ubuntu 20.04)

### ISSUE THREAD

**Issue by lanctot** [FloopCZ/tensorflow_cc#218]
Seems like Ubuntu 20.04 does not provide the `python` itself (only `python3`), so bazel fails when looking for it:

```
INFO: Analyzed 2 targets (184 packages loaded, 16898 targets configured).
INFO: Found 2 targets...
INFO: Deleting stale sandbox base /home/lanctot/.cache/bazel/_bazel_lanctot/7506f1f517861b11f55b320df18b36b9/sandbox
ERROR: /home/lanctot/tensorflow_cc/tensorflow_cc/build/tensorflow/tensorflow/core/util/BUILD:345:1: Executing genrule //tensorflow/core/util:version_info_gen failed (Exit 127)
/usr/bin/env: 'python': No such file or directory
INFO: Elapsed time: 14.947s, Critical Path: 0.08s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
make[2]: *** [CMakeFiles/tensorflow_base.dir/build.make:115: tensorflow-stamp/tensorflow_base-build] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/tensorflow_base.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
lanctot@nife:~/tensorflow_cc/tensorflow_cc/build$ which python
lanctot@nife:~/tensorflow_cc/tensorflow_cc/build$ which python3
/usr/bin/python3
lanctot@nife:~/tensorflow_cc/tensorflow_cc/build$ 
```

I wonder if cmake can check for `python` and alias it to `python3` if it's not found?

**Comment by lanctot** [comment]
I tried with aliasing python to python3, but no dice.

I think this is related to using an older version of bazel, i.e. 2.0.0 (installed because make would fail otherwise, see: https://github.com/FloopCZ/tensorflow_cc/issues/216), which is probably looking for python 2...?

So I'm not sure how to fix this. I know they added Python 3.8 support in TF2.2 so it should not be a problem. But maybe they have not tested on Ubunut 20.04 yet.

**Comment by sillick-rfa** [comment]
That's odd because the environment variables set in the build_tensorflow.sh file look for and set it to python3

```
export PYTHON_BIN_PATH=${PYTHON_BIN_PATH:-"$(which python3)"}
export PYTHON_LIB_PATH="$($PYTHON_BIN_PATH -c 'import site; print(site.getsitepackages()[0])')"
```

**Comment by lanctot** [comment]
Yes, and also I see this output too when it starts building:

```
INFO: Reading rc options for 'build' from /home/lanctot/tensorflow_cc/tensorflow_cc/build/tensorflow/.tf_configure.bazelrc:
  'build' options: --action_env PYTHON_BIN_PATH=/usr/bin/python3 --action_env PYTHON_LIB_PATH=/usr/local/lib/python3.8/dist-packages --python_path=/usr/bin/python3 --config=xla --action_env TF_CONFIGURE_IOS=0
```

So clearly it's finding it and sending it to bazel, but then down the road it still crashes. I tried defining python runtimes and adding flags to the bazel build that I found from a few searches, and nothing worked. It finally progressed when I manually added the symlink:

```
sudo ln -s /usr/bin/python3 /usr/bin/python
```

Seems to me that this is a Tensorflow issue. The genrule target //tensorflow/core/util:version_info_gen is assuming python is in the path, but that is no longer true by default in Ubuntu 20.04. I guess they just haven't tested on Ubuntu 20.04 yet. So I will close this issue.

But, my suggestion would be to add a check in `build_tensorflow.sh` to see if `python` is in the path, and if not, then exit with a warning that `python` needs to be in the PATH otherwise the TF build fails.
